src/tree-key-cache.ts

```diff
@@ -156,4 +179,110 @@ export class TreeKeyCache<T> {
 			}
 		}
 	}
+
+	/**
+	 * Set all the given values of the informed tree
+	 * @param tree The tree with the values to be set. It need to start from root node. Any node you don't want to change, just leave it undefined
+	 * @param createValue Callback for leaf creation when it does not exist
+	 */
+	async *fullTreeSet(
+		tree: Tree<T>,
+		createValue: () => T,
+	): AsyncIterable<FullSetItem<T>> {
+		const iterable = treePreOrderBreadthFirstSearch(tree);
```

Comment:
The iteration for every node that is saved plain on redis, ie, without falling into the key with a tree structure, must be made using a breadth first search traversal algorithm, as it minimizes the memory needed to keep tracking of the keys